### PR TITLE
bring in python devshell with Nix

### PR DESCRIPTION
**Pull request by flokli**
This drops a `python.withPackages (…)` with the devshell, and reconfigures pytest to add `.` to `PYTHONPATH`.

This removes the additional `pip install` run whenever the shell is entered, so all targets and scripts related to managing that can be removed.

`pytest` now uses that `python` for all third-party python dependencies - which required exposing `pyrate-limiter` and `requests-limiter`.

Some of the tests were previously "integration-testing the flake outputs", introducing additional complexity and another nix eval - we run the tests inside the dev environment, so there's no need to do that.

If we really want to test the flake outputs still work, it could be a separate "smoketest" (maybe just inside GH actions). `pytest` should not execute Nix (at least where we can avoid it, which is bringing in the environment we already have).

---

Changing to this approach exposed some smell - the entrypoints are not really thin entrypoints, rely on and execute other binaries in `$PATH`, too.

Some of this code should probably be moved to some library code, most pressingly, `vulnxscan` should not invoke `repology_cli` (which is another entrypoint), but instead repology-related code should be moved into a lib that can be used by both consumers.

For now, I disabled these three (now failing) pytest tests. In nix-based builds, this doesn't matter and will still work, as we ship all entrypoints in a single `$out/bin`. 

--

Long-term, we might want to decouple the "CLI parsing from process args and preparing result for presentation" part from the actually code that run. (Ideally by moving the "main logic" into lib code too).

Rather than starting another python interpreter and passing CLI args, we could then really test the individual primitives, which should allow more fine-grained testing than just "integration-testing" the whole CLI, which is what we do now.

### CODE REVIEW

File: nix/checks.nix [hunk 1]
```diff
@@ -9,16 +9,44 @@
   }: {
     checks =
       {
+        black =
+          pkgs.runCommandLocal "black" {
+            nativeBuildInputs = [pkgs.python3.pkgs.black];
+          } ''
+            cd ${../.}
```
Comment (flokli) [inline nix/checks.nix:16]: @brianmcgee We're using this quite frequently - it'd probably make sense to construct a gitingore-cleaned src once in a let binding and then just pass it down to all checks (or even doing the filtering we currently do at runtime in Nix), hopefully resulting in less rebuilds.

File: nix/checks.nix [hunk 2]
```diff
@@ -9,16 +9,44 @@
...
+            for py in $(find . -name "*.py" ! -path "*venv*" ! -path "*eggs*"); do
```
Comment (flokli) [inline nix/checks.nix:17]: All the eggs and venvs are gone, we can probably drop this (and the `make clean` target) in a followup.

File: nix/checks.nix (hunk 1 above)
Comment (brianmcgee) [reply nix/checks.nix:16]: @flokli You can enable black in treefmt and remove this. There's already a treefmt check registered.

Comment (flokli) [reply nix/checks.nix]: pushed your commit to my branch, thanks!

**Review by brianmcgee — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -71,12 +71,22 @@ $ cd sbomnix
...
+Keep in mind this doesn't add the various entrypoint binaries to your `PATH`
+directly. They are produced during the setuptools build.
+
+While you're in the devshell, you can run run `sbomnix` via the entrypoint file
+directly:
+
 ```bash
-$ sbomnix --help
+$ sbomnix/main.py --help
 ```
```
Comment (henrirosten) [inline README.md:82]: I think we need to overview the documentation across the repo, and make sure it instructs correct usage.
We (I) can also do that in the follow-up PR later.

**Review by henrirosten — APPROVED** [review]

File: nix/checks.nix (hunk 2 above)
Comment (flokli) [reply nix/checks.nix]: Marked as resolved for now, let's leave this to a followup.

**Pull request merged**